tests/unit/layers/models/layersManager.spec.ts

```diff
@@ -0,0 +1,717 @@
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+
+      await expect(action).rejects.toThrow(BadRequestError);
+      expect(fileValidatorValidateExistsMock).toHaveBeenCalledTimes(1);
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(0);
+      expect(findJobsMock).toHaveBeenCalledTimes(0);
+      expect(createSplitTilesTasksMock).toHaveBeenCalledTimes(0);
+    });
+
+    // TODO: Handle test when update is supported for other formats
+    it('should throw Bad Request Error for "Update" job type if there is unsupported file (not GPKG) in request', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1,2-3' });
+      setValue('ingestionTilesSplittingTiles.tasksBatchSize', 2);
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      getHighestLayerVersionMock.mockResolvedValue(2.5);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([]);
+      validateGpkgFilesMock.mockReturnValue(false);
+      createLayerJobMock.mockResolvedValue('testJobId');
+      createMergeTilesTasksMock.mockResolvedValue(undefined);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
+      // layersManager = new LayersManager(
+      //   logger,
+      //   zoomLevelCalculator,
+      //   jobManagerClientMock,
+      //   catalogClientMock,
+      //   mapPublisherClientMock,
+      //   fileValidatorMock,
+      //   splitTilesTaskerMock,
+      //   mergeTilesTasker
+      // );
+
+      const action = async () => {
+        await layersManager.createLayer(testData, managerCallbackUrl);
+      };
+
+      await expect(action).rejects.toThrow(BadRequestError);
+      expect(getHighestLayerVersionMock).toHaveBeenCalledTimes(1);
+    });
+
+    it('fail if layer status is pending', async function () {
+      // eslint-disable-next-line @typescript-eslint/naming-convention
+      setValue({ 'tiling.zoomGroups': '1' });
+      const testData: IngestionParams = {
+        fileNames: ['test.tif'],
+        metadata: { ...testImageMetadata },
+        originDirectory: '/here',
+      };
+
+      catalogExistsMock.mockResolvedValue(false);
+      fileValidatorValidateExistsMock.mockResolvedValue(true);
+      validateSourceDirectoryMock.mockResolvedValue(true);
+      validateNotWatchDirMock.mockResolvedValue(true);
+      findJobsMock.mockResolvedValue([{ status: OperationStatus.PENDING }]);
+
+      // const zoomLevelCalculator = new ZoomLevelCalculator(configMock);
```

Comment:
why the remarks?

Reply:
moved into before each test instad of create new instance every test manualy